src/Microsoft.OData.Core/ODataLibraryCompatibility.cs

```diff
@@ -4,29 +4,49 @@
 // </copyright>
 //---------------------------------------------------------------------
 
-using System.Diagnostics.CodeAnalysis;
+using System;
 
 namespace Microsoft.OData
 {
     /// <summary>
     /// Library compatibility levels.
     /// </summary>
-    [SuppressMessage("Microsoft.Design", "CA1008:EnumsShouldHaveZeroValue")]
+    [Flags]
     public enum ODataLibraryCompatibility
     {
         /// <summary>
-        /// Version 6.x
+        /// No backward compatibility feature flag set.
         /// </summary>
-        Version6 = 060000,
+        None = 0,
 
         /// <summary>
-        /// Version 7.x
+        /// When enabled, the CSDL Scale and SRID "variable" value will be written "Variable" instead of "variable" for compatibility with older versions of the library.
+        /// </summary>
+        UseLegacyVariableCasing = 1 << 0,
+
+        /// <summary>
+        /// When enabled, writes a null odata annotation.@odata.null=true
+        /// </summary>
+        WriteNullODataAnnotation = 1 << 1,

```

Comment:
@ElizabethOkerio The name for this enum member `WriteNullODataAnnotation` is not intuitive. Firstly from the fact that `Null` comes ahead of `OData` which kind of suggests that the name could describe any of these examples:
- `"@odata.id": null`
- `"@odata.removed": null`
- etc.

For the reason that they are all OData annotations with a value of null - `NullODataAnnotation`.

Going by the [message for the exception](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/Microsoft.OData.Core.txt#L89) we throw in this scenario, I'd suggest we name this `WriteTopLevelODataNullAnnotation` for the followings reasons:
- It's referred to as `ODataNull` - [`ODataAnnotationNames.ODataNull`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/ODataAnnotationNames.cs#L108) and [`JsonConstants.PrefixedODataNullPropertyName`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/JsonLightConstants.cs#L130)
- The particular use case that the flag controls is writing top-level property - [`IsTopLevel6xNullValue`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs#L2123), [`IsTopLevel6xNullValueAsync`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs#L3508), [`WriteNullProperty`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs#L647), and [`WriteNullPropertyAsync`](https://github.com/OData/odata.net/blob/24ae2e335650c2533e541e1956f5b64f618ee0cd/src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs#L1086)